tcms/testplans/tests/test_new_plan.py

```diff
@@ -76,36 +118,117 @@ def test_plan_create_new_inactive(self):
         self.assertTrue(plan.emailing.auto_to_plan_author)
         self.assertTrue(plan.emailing.auto_to_case_owner)
         self.assertTrue(plan.emailing.auto_to_case_default_tester)
         self.assertTrue(plan.emailing.notify_on_plan_update)
         self.assertTrue(plan.emailing.notify_on_case_update)
 
-    def test_get_with_no_perm_redirects_to_login(self):
-        remove_perm_from_user(self.user, self.add_testplan_permission)
+    def test_with_invalid_product_shows_error(self):
+        new_data = self.request.copy()
+        new_data["product"] = -1
+        del new_data["email_settings-0-auto_to_plan_author"]
+
+        response = self.client.post(self.url, data=new_data, follow=True)
+
+        self.assertContains(response, _("Create new TestPlan"))
+        self.assertContains(
+            response,
+            _(
+                "Select a valid choice. That choice is not one of the available choices."
+            ),
+        )
+        self.assertContains(
+            response,
+            '<input class="bootstrap-switch" name="email_settings-0-auto_to_plan_author" '
+            'type="checkbox"',
+            html=False,
+        )
+
+
+class TestNewTestPlanViewPermissions(PermissionsTestCase):
+    add_testplan_permission = "testplans.add_testplan"
+    permission_label = add_testplan_permission
+    http_method_names = ["get", "post"]
+    url = reverse("plans-new")
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.post_data = {"description": "test"}
+        super().setUpTestData()
+
+        cls.product = ProductFactory()
+        cls.product_version = VersionFactory(product=cls.product)
+        cls.test_plan_type = PlanTypeFactory()
+        cls.test_plan_1 = TestPlanFactory()
+
+        cls.post_data.update(
+            {
+                "author": cls.tester.pk,
+                "product": cls.product.pk,
+                "product_version": cls.product_version.pk,
+                "type": cls.test_plan_type.pk,
+                "name": cls.test_plan_1.name,
+                "email_settings-0-auto_to_plan_author": "on",
+                "email_settings-0-auto_to_case_owner": "on",
+                "email_settings-0-auto_to_case_default_tester": "on",
+                "email_settings-0-notify_on_case_update": "on",
+                "email_settings-0-notify_on_plan_update": "on",
+                "email_settings-0-id": cls.test_plan_1.emailing.pk,
+                "email_settings-TOTAL_FORMS": "1",
+                "email_settings-INITIAL_FORMS": "1",
+                "email_settings-MIN_NUM_FORMS": "0",
+                "email_settings-MAX_NUM_FORMS": "1",
+                "is_active": True,
+            }
+        )
+
+    def verify_get_with_permission(self):
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, _('Create new TestPlan'))
+
+    def verify_post_with_permission(self):
```

Comment:
I can't figure out why this is failing

Reply:
because self.tester is missing testplans.view_testplan permission

Reply:
I can't request a review so pinging @atodorov